### ISSUE TITLE
menuContainer not working

### ISSUE THREAD

**Issue by Alex61NN5** [ladderio/ngx-tribute#13]
Hey There,

I am trying to use this package in my application, basically everything works great except for `menuContainer`

My component is set up like so 

**HTML**
```
<div class="container">
    <textarea [ngxTribute]="options"></textarea>
</div>
```

**.TS**
```
values = [
 ...
]
options: TributeOptions<any> = {
    values: this.values,
    menuItemTemplate: () => {
        return `...`
    },
   menuContainer: document.querySelector('.container'),
   positionMenu: false
}
```

now what is happening is that, the `ngxTributeDirective` is being loaded correctly, but it is not honoring the parent container specified in the `menuContainer` value.

Now I think it may be because, the page has not yet rendered the `.container` html element before the `ngxTribute` is initialised, so it skips it and the tribute container does not get a parent container

I used the es6 version of tribute and had to declare the tribute object `AfterViewInit` and it worked.

I tried to create a stackblitz of the issue but the package would not install correctly on there system.

To recreate this issue the easiest way would to be put your input inside a parent container and set the menuContainer value to the parent container class and `positionMenu: false`

**Comment by Alex61NN5** [comment]
This could potentially be solved by moving, 

```
this.tribute = new Tribute(this.options);
this.tribute.attach(this.element.nativeElement);
this.element.nativeElement.addEventListener('tribute-replaced', () => {
    const value = ['INPUT', 'TEXTAREA'].includes(
        this.element.nativeElement.tagName
    )
        ? this.element.nativeElement.value
        : this.element.nativeElement.innerHTML;

        this.onMentioned.emit(value);

        if (this.control) {
            this.control.setValue(value);
        }
});
```

from `ngOnInit()` to `ngAfterViewInit()`

**Comment by Alex61NN5** [comment]
After some testing it does not seem `ngAfterViewInit` or `ngAfterViewChecked()` fix this issue

**Comment by agarbund** [comment]
Moving the code to `ngAfterViewInit()` should do the job.

I think the problem is caused by creating options for Tribute before `.container'` element is visible. `document.querySelector('.container')` is evaluated during options object creation, and returns `null` there. This object is then passed to the directive, but as `menuContainer` was already evaluated to `null` it doesn't matter if Tribute will be initialized in `ngOnInit()` or `ngAfterViewInit()`.

The solution could be allowing to pass a function which resolves to the options object, not the object itself. This way initialization of options could truly be done in `ngAfterViewInit()` by calling this function there.

I'll try to tackle this over the weekend.

**Comment by Alex61NN5** [comment]
Thank you!

**Comment by Alex61NN5** [comment]
@agarbund were you able to solve this issue??

**Comment by agarbund** [comment]
Fixed in version 1.3.0. See the updated docs https://github.com/ladderio/ngx-tribute#using-menucontainer
